### PR TITLE
Testing of the ConfirmedLesson screen

### LINKED ISSUE #108: Write tests for ConfirmedLesson.kt

### PR DESCRIPTION
**Pull request by Philipp0206**
PR Description
This PR aims to test the ConfirmedLesson screen that was implemented in a previous PR (#95). 

Tests Description
- The first test checks if everything is displayed on the screen when we click on a confirmed lesson.
- The 2nd if the back button goes indeed back to the previous screen if pushed.
- The 3rd checks SMS app opening.
- The 4th checks the case when no profile is found and if indeed there is a message that shows up.
- The last checks if no Lesson is selected and if indeed there is a message that shows up.

Close #108

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import org.mockito.Mockito.*
+import org.mockito.kotlin.whenever
+
+@RunWith(AndroidJUnit4::class)
+public class ConfirmedLessonTest {
+
+  @get:Rule val composeTestRule = createComposeRule()
+  @get:Rule
+  val permissionRule: GrantPermissionRule =
+      GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
```
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt:31]: Nice way to simulate location granting!

File: app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+  @Before
+  fun setUp() {
+    whenever(mockProfilesRepository.getProfiles(org.mockito.kotlin.any(), org.mockito.kotlin.any()))
+        .thenAnswer { invocation ->
+          val onSuccess = invocation.arguments[0] as (List<Profile>) -> Unit
+          onSuccess(
+              listOf(
+                  tutorProfile,
+                  studentProfile)) // Simulate a list of profiles with our beloved Ozymandias
```
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt:86]: Very thorough mocking!

File: app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        }
+
+    whenever(
+            mockLessonRepository.getLessonsForTutor(
+                org.mockito.kotlin.eq(tutorProfile.uid),
```
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt:91]: Good idea looking for the specific profile in the mock

File: app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+          val onSuccess = invocation.arguments[1] as (List<Lesson>) -> Unit
+          onSuccess(listOf(lesson))
+        }
+
+    listProfilesViewModel.getProfiles()
+    lessonViewModel.getLessonsForTutor(tutorProfile.uid, {})
+
+    lessonViewModel.selectLesson(lesson)
+    listProfilesViewModel.setCurrentProfile(tutorProfile)
+    Log.e("LeProfileBG", "${listProfilesViewModel.currentProfile}")
```
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt:104]: Maybe remove this log for next time

File: app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+  @Test
+  fun confirmedLessonScreenBackButtonClicked() {
+    composeTestRule.setContent {
+      ConfirmedLessonScreen(
+          listProfilesViewModel = listProfilesViewModel,
+          lessonViewModel = lessonViewModel,
+          navigationActions = mockNavigationActions)
+    }
+    composeTestRule.waitForIdle()
```
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt:131]: Good idea, should be a significant improvement over guesstimated sleep calls

File: app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+  fun confirmedLessonScreenOpensSmsApp() {
+    composeTestRule.setContent {
+      ConfirmedLessonScreen(
+          listProfilesViewModel = listProfilesViewModel,
+          lessonViewModel = lessonViewModel,
+          navigationActions = mockNavigationActions)
+    }
+
+    // Click on the "Message Tutor/Student" button
+    composeTestRule.onNodeWithTag("contactButton").performClick()
```
Comment (A-M-00) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt:147]: This test doesn't seem to actually test anything. is it a placeholder?

File: app/src/main/java/com/github/se/project/model/profile/ListProfilesViewModel.kt [hunk 7]
```diff
@@ -104,7 +104,7 @@ open class ListProfilesViewModel(private val repository: ProfilesRepository) : V
    *
    * @param profile The Profile document to be selected.
    */
-  fun setCurrentProfile(profile: Profile) {
+  fun setCurrentProfile(profile: Profile?) {
```
Comment (A-M-00) [inline app/src/main/java/com/github/se/project/model/profile/ListProfilesViewModel.kt:107]: I don't really see why this is necessary, but if it helps the testing it shouldn't be harmful either

**Review by A-M-00 — APPROVED** [review]
Pretty good job overall! Effective testing that should increase out coverage nicely

File: app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt (hunk 4 above)
Comment (Philipp0206) [reply app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt:104]: You're right but for CI reasons I want to fix this next time or tell to a teammate to remove it in his PR, as by mircale I finally pass the CI.

Comment (Philipp0206) [reply app/src/androidTest/java/com/github/se/project/ui/lesson/ConfirmedLessonTest.kt:104]: Will be fixed soon in Léa's PR.

**Pull request merged**